src/couch_multidb_changes.erl

```diff
@@ -282,3 +282,472 @@ is_design_doc_id(<<?DESIGN_DOC_PREFIX, _/binary>>) ->
 
 is_design_doc_id(_) ->
     false.
+
+
+
+-ifdef(TEST).
+
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("couch/include/couch_eunit.hrl").
+
+-define(MOD, multidb_test_module).
+-define(SUFFIX, <<"suff">>).
+-define(DBNAME, <<"shards/40000000-5fffffff/acct/suff.0123456789">>).
+
+couch_multidb_changes_test_() ->
+    {
+        foreach,
+        fun setup/0,
+        fun teardown/1,
+        [
+            t_handle_call_change(),
+            t_handle_call_change_filter_design_docs(),
+            t_handle_call_created(),
+            t_handle_call_deleted(),
+            t_handle_call_checkpoint_new(),
+            t_handle_call_checkpoint_existing(),
+            t_handle_info_scanner_exit_normal(),
+            t_handle_info_scanner_crashed(),
+            t_handle_info_event_listener_exited(),
+            t_handle_info_unknown_pid_exited(),
+            t_handle_info_change_feed_exited(),
+            t_handle_info_change_feed_exited_and_need_rescan(),
+            t_spawn_changes_reader(),
+            t_changes_reader_cb_change(),
+            t_changes_reader_cb_stop(),
+            t_changes_reader_cb_other(),
+            t_handle_call_resume_scan_no_chfeed_no_ets_entry(),
+            t_handle_call_resume_scan_chfeed_no_ets_entry(),
+            t_handle_call_resume_scan_chfeed_ets_entry(),
+            t_handle_call_resume_scan_no_chfeed_ets_entry(),
+            t_start_link(),
+            t_start_link_no_ddocs(),
+            t_handle_db_event_created(),
+            t_handle_db_event_deleted(),
+            t_handle_db_event_updated(),
+            t_handle_db_event_other(),
+            t_scanner_finds_shard(),
+            t_misc_gen_server_callbacks()
+        ]
+    }.
+
+setup() ->
+    mock_logs(),
+    mock_callback_mod(),
+    meck:expect(couch_event, link_listener, 4, {ok, elpid}),
+    meck:expect(config, get, fun("couchdb", "database_dir", _) -> ?TEMPDIR end),
+    mock_changes_reader().
+
+teardown(_) ->
+    meck:unload(),
+    delete_shard_file(?DBNAME).
+
+
+t_handle_call_change() ->
+    ?_test(begin
```

Comment:
?_test` and `begin/end` is used because top level test is a generator and test list is expected to be a list of test generators. The other pattern is to put `?_test` in the top level generator list like so `[…, ?_test(t_handle_call_change()),…]   …. t_handle_call_change()-> setup…, ?assert(…).` This avoids needing to add `?_test(begin…end)` inside each function